components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +93,297 @@
+                                .get(APIConstants.SWAGGER_X_SCOPES_BINDINGS)) != null) {
+                    if (scopeBindings.get(scope.getKey()) != null) {
+                        scope.setRoles(scopeBindings.get(scope.getKey()));
+                    }
+                }
+                scopeSet.add(scope);
+            }
+            setOtherSchemes(otherSetOfSchemes);
+            return OASParserUtil.sortScopes(scopeSet);
+        } else {
+            setOtherSchemes(otherSetOfSchemes);
+            return OASParserUtil.sortScopes(getScopesFromExtensions(swagger));
+        }
+    }
+
+    /**
+     * This method returns URI templates according to the given swagger file(Swagger version 2)
+     *
+     * @param resourceConfigsJSON swaggerJSON
+     * @return URI Templates
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<URITemplate> injectOtherResourceScopesToDefaultScheme(String resourceConfigsJSON) throws APIManagementException {
+        Swagger swagger = getSwagger(resourceConfigsJSON);
+        Set<URITemplate> urlTemplates = new LinkedHashSet<>();
+        Set<Scope> scopes = injectOtherScopesToDefaultScheme(resourceConfigsJSON);
+        List<String> schemes = getOtherSchemes();
+        boolean isDefaultAvailable = isDefaultGiven(resourceConfigsJSON);
+
+        for (String pathString : swagger.getPaths().keySet()) {
+            Path path = swagger.getPath(pathString);
+            Map<HttpMethod, io.swagger.models.Operation> operationMap = path.getOperationMap();
+            for (Map.Entry<HttpMethod, io.swagger.models.Operation> entry : operationMap.entrySet()) {
+                io.swagger.models.Operation operation = entry.getValue();
+                URITemplate template = new URITemplate();
+                template.setHTTPVerb(entry.getKey().name().toUpperCase());
+                template.setHttpVerbs(entry.getKey().name().toUpperCase());
+                template.setUriTemplate(pathString);
+                List<String> opScopesofDefault = getScopeOfOperations(SWAGGER_SECURITY_SCHEMA_KEY, operation);
+                if (opScopesofDefault == null) {
+                    opScopesofDefault = new ArrayList<>();
+                }
+                //Handling scopes in resources which do not belong to 'default' scheme
+                if (!isDefaultAvailable) {
+                    for (int i = 0; i < schemes.size(); i++) {
+                        List<String> opScopesOfOthers = getScopeOfOperations(schemes.get(i), operation);
+                        if (!opScopesOfOthers.isEmpty()) {
+                            if (opScopesOfOthers.size() == 1) {
+                                String firstScope = opScopesOfOthers.get(0);
+                                Scope scope = APIUtil.findScopeByKey(scopes, firstScope);
+                                if (scope == null) {
+                                    throw new APIManagementException("Scope '" + firstScope + "' not found.");
```

Comment:
```suggestion
                                    throw new APIManagementException("Scope '" + firstScope + "' not found. Probably it is not defined in the securityDefinitions of the provided OAS definition");
```

Did you want to mean the above?

Reply:
Yeah

Reply:
Also, this is a user error, it is better to give this as an error response of the swagger update/import REST API call. For this, we have to define a custom exception code and use it when throwing the APIManagementException. Please refer the wiki page [Exception Handling in REST APIs](https://github.com/wso2/carbon-apimgt/wiki/Exception-handling-in-REST-APIs)'